common/lib/xmodule/xmodule/js/karma_xmodule.conf.js

```diff
@@ -79,7 +80,8 @@ var options = {
 
     fixtureFiles: [
         {pattern: 'fixtures/*.*'},
-        {pattern: 'fixtures/hls/**/*.*'}
+        {pattern: 'fixtures/hls/**/*.*'},
+        {pattern: 'common_static/common/templates/video/autoadvance-button.underscore'}
```

Comment:
@clemente Since `common_static` is a symlink to [`../../../../static`](https://github.com/open-craft/edx-platform/blob/8968f3d71117c01079c8ea0d65679f4f1dcdd7b1/common/lib/xmodule/xmodule/js/common_static#L1), this template path might be normalizing relative to the base path differently between karma and requirejs (the symlink might not be getting flattened for karma). Could you try changing it to `../../../../static/common/templates/video/autoadvance-button.underscore`, and paste the new 404 output (if any)? We might have to dig into the specifics of how the paths are getting resolved [here](https://github.com/open-craft/edx-platform/blob/8968f3d71117c01079c8ea0d65679f4f1dcdd7b1/common/static/common/js/karma.common.conf.js) to figure it out.

Reply:
@bdero Good idea, I tried but I saw the same error. Then I tested changing the path [where the underscore file is used](https://github.com/edx/edx-platform/pull/15803/files#diff-912c26eca052b61c3e644aac9255347eR7), and this brought me to different error messages:
I tried your value, and the 404 was `Error: /edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/js/video/../../../../static/common/templates/video/autoadvance-button.underscore HTTP status: 404`. This is because it's missing an extra `../`.
With five `../` the error was `Error: /edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/js/video/../../../../../static/common/templates/video/autoadvance-button.underscore HTTP status: 404`. I noticed that it didn't work because of the `js/video` part, namely in `js` there's no `video` folder (probably due to [this code](https://github.com/open-craft/edx-platform/blob/8968f3d71117c01079c8ea0d65679f4f1dcdd7b1/common/static/common/js/karma.common.conf.js#L43)). I created a `video` folder there. Then the error is still the same (404) but the displayed path does exist and resolve to a file:
```
ll /edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/js/video/../../../../../static/common/templates/video/autoadvance-button.underscore
-rw-r--r-- 1 syslog edxadmin 178 Nov 28 23:15 /edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/js/video/../../../../../static/common/templates/video/autoadvance-button.underscore
```
So now there's no reason for it not to find the file…
My hypothesis is that the part of `text.js` which is failing (see `at /edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/js/common_static/js/vendor/requirejs/text.js:294` in error) is trying to [do an XMLHttpRequest](https://github.com/edx/edx-platform/blob/master/common/static/js/vendor/requirejs/text.js#L270) and it might not support descending directories. I can debug this more in `text.js`.